src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,255 @@
+import contextlib
+import errno
+import importlib.resources as pkg_resources
+import json
+import logging
+import os
+import pathlib
+import subprocess
+import sys
+import tempfile
+import traceback
+import urllib.request
+from typing import Generator, List, Optional, Tuple
+import importlib.resources as pkg_resources
+
+import click
+
+if sys.platform.startswith("win"):
+    import winreg
+
+_logger = logging.getLogger(__name__)
+
+METADATA_FILE = "_installer_metadata.json"
+
+
+
+def _parse_version(version: str) -> Tuple[int, ...]:
+    """
+    Split the version string into a tuple of integers.
+
+    >>> _parse_version("23.8.0")
+    (23, 8, 0)
+    >>> _parse_version("24.0.0")
+    (24, 0, 0)
+    >>> _parse_version("invalid_version")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.ClickException: Invalid version format found
+    """
+    try:
+        return tuple(int(part) for part in version.split("."))
+    except ValueError as e:
+        _logger.info("Failed to parse version.", exc_info=True)
+        raise click.ClickException(f"Invalid version format found:{str(e)}") from e
+
+
+def _get_daqmx_installed_version() -> Optional[str]:
+    """
+    Check for existing installation of NI-DAQmx.
+
+    """
+    try:
+        _logger.debug("Reading the registry entries to get installed DAQmx version")
+        with winreg.OpenKeyEx(
```

Comment:
All `winreg` access should be inside an `if sys.platform == "win32":` block. The else case can raise an exception.

mypy caught this:
```
generated/nidaqmx/_install_daqmx.py:54: error: Name "winreg" is not defined  [name-defined]
generated/nidaqmx/_install_daqmx.py:55: error: Name "winreg" is not defined  [name-defined]
generated/nidaqmx/_install_daqmx.py:58: error: Name "winreg" is not defined  [name-defined]
generated/nidaqmx/_install_daqmx.py:60: error: Name "winreg" is not defined  [name-defined]
generated/nidaqmx/_install_daqmx.py:61: error: Name "winreg" is not defined  [name-defined]
```